src/utils/slicer.py

```diff
@@ -34,4 +54,19 @@ def __call__(self, text: str, max_slice_count: int = None) -> torch.Tensor:
         image_bytes = 255 - image_bytes
         slices = image_bytes.unfold(1, self.window_size, self.stride).permute((1, 0, 2))[:max_slice_count]
 
+        lb = 0
+        rb = min(self.window_size - 1, len(char_num) - 1)
+        r_shift = 1 if char_ratio_r[rb] >= self.ratio else 0
+        slice_text = [text[char_num[lb] : char_num[rb] + r_shift]]
+
+        for i in range(len(slices) - 1):
+            lb += self.stride
+            rb += self.stride
+            if rb >= len(char_num):
+                rb = len(char_num) - 1
+
+            l_shift = 0 if char_ratio_l[lb] >= self.ratio else 1
+            r_shift = 1 if char_ratio_r[rb] >= self.ratio else 0
+            slice_text.append(text[char_num[lb] + l_shift : char_num[rb] + r_shift])
+
         return slices
```

Comment:
Why don't you return texts? It seems you extracted them into `slice_text` and didn’t use them.